lib/Mojo/Promise.pm

```diff
@@ -215,7 +215,8 @@ sub _settle {
   }
 
   if ($thenable && $status eq 'resolve') {
-    $results[0]->then(sub { $self->resolve(@_); () }, sub { $self->reject(@_); () });
+    if (refaddr $results[0] == refaddr $self) {$self->reject('Chaining cycle detected')}
+    else { $results[0]->then(sub { $self->resolve(@_); () }, sub { $self->reject(@_); () }); }
```

Comment:
I don't think this is run through perltidy. I suggest this formatting:

```
    (refaddr $results[0] == refaddr $self)
      ? $self->reject('Chaining cycle detected.')
      : $results[0]->then(sub { $self->resolve(@_); () }, sub { $self->reject(@_); () });
```

Reply:
I don't have access to perltidy right now. You'd have to run it through perltidy yourself, if that's not a huge problem for you. Thanks.

Reply:
addressed